sql/api/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -315,6 +317,10 @@ case class StructType(fields: Array[StructField]) extends DataType with Seq[Stru
     nameToIndex.get(name)
   }
 
+  def getFieldIndexCaseInsensitive(name: String): Option[Int] = {
```

Comment:
Let's don't add this as an API tho.

```suggestion
  private[sql] def getFieldIndexCaseInsensitive(name: String): Option[Int] = {
```